porch/engine/pkg/kpt/fs_test.go

```diff
@@ -34,3 +40,223 @@ func TestMemFS(t *testing.T) {
 		t.Errorf("unexpected file contents: got %q, want %q", got, want)
 	}
 }
+
+func TestMemFSRenderBasicPipeline(t *testing.T) {
+	resources := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+---
+apiVersion: custom.io/v1
+kind: Custom
+metadata:
+  name: custom
+spec:
+  image: nginx:1.2.3`
+
+	kptfile := `apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: app
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/set-namespace:v0.1.3
+      configMap:
+        namespace: staging
+    - image: gcr.io/kpt-fn/set-labels:v0.1.4
+      configMap:
+        tier: backend`
+
+	expectedResources := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: staging
+  labels:
+    tier: backend
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      tier: backend
+  template:
+    metadata:
+      labels:
+        tier: backend
+---
+apiVersion: custom.io/v1
+kind: Custom
+metadata:
+  name: custom
+  namespace: staging
+  labels:
+    tier: backend
+spec:
+  image: nginx:1.2.3
+`
+
+	fs := &memfs{}
+	if err := fs.MkdirAll("a/b/c"); err != nil {
+		t.Errorf(`MkdirAll("a") failed %v`, err)
```

Comment:
nit: MkdirAll("a/b/c/") ? (I would not be surprised it if it my bug :)

Reply:
Updated. Thanks